rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -0,0 +1,356 @@
+        if os.path.isfile(featurizer_file):
+            return jsonpickle.decode(rasa.utils.io.read_file(featurizer_file))
+
+        logger.error(
+            "Couldn't load featurizer for policy. "
+            "File '{}' doesn't exist.".format(featurizer_file)
+        )
+        return None
+
+
+class FullDialogueTrackerFeaturizer(TrackerFeaturizer):
+    """Creates full dialogue training data for time distributed architectures.
+
+    Creates training data that uses each time output for prediction.
+    Training data is padded up to the length of the longest dialogue with -1.
+    """
+
+    def training_states_and_actions(
+        self, trackers: List[DialogueStateTracker], domain: Domain
+    ) -> Tuple[List[List[State]], List[List[Text]]]:
+        """Transforms list of trackers to lists of states and actions.
+
+        Training data is padded up to the length of the longest dialogue with -1.
+        """
+
+        trackers_as_states = []
+        trackers_as_actions = []
+
+        logger.debug(
+            "Creating states and action examples from "
+            "collected trackers (by {}({}))..."
+            "".format(type(self).__name__, type(self.state_featurizer).__name__)
+        )
+        pbar = tqdm(trackers, desc="Processed trackers", disable=is_logging_disabled())
+        for tracker in pbar:
+            states = self._create_states(tracker, domain)
+
+            delete_first_state = False
+            actions = []
+            for event in tracker.applied_events():
+                if isinstance(event, ActionExecuted):
+                    if not event.unpredictable:
+                        # only actions which can be
+                        # predicted at a stories start
+                        actions.append(event.action_name or event.action_text)
+                    else:
+                        # unpredictable actions can be
+                        # only the first in the story
+                        if delete_first_state:
+                            raise Exception(
+                                "Found two unpredictable "
+                                "actions in one story."
+                                "Check your story files."
+                            )
+                        else:
+                            delete_first_state = True
+
+            if delete_first_state:
+                states = states[1:]
+
+            trackers_as_states.append(states[:-1])
+            trackers_as_actions.append(actions)
+
+        return trackers_as_states, trackers_as_actions
+
+    def prediction_states(
+        self, trackers: List[DialogueStateTracker], domain: Domain
+    ) -> List[List[State]]:
+        """Transforms list of trackers to lists of states for prediction."""
+
+        trackers_as_states = [
+            self._create_states(tracker, domain) for tracker in trackers
+        ]
+
+        return trackers_as_states
+
+
+class MaxHistoryTrackerFeaturizer(TrackerFeaturizer):
```

Comment:
is that also new or just copied?

Reply:
It's modified, but the featurizer was always present